sql/core/src/test/scala/org/apache/spark/sql/execution/vectorized/ColumnarBatchSuite.scala

```diff
@@ -1688,6 +1688,45 @@ class ColumnarBatchSuite extends SparkFunSuite {
     }
   }
 
+  test("SPARK-37161: RowToColumnConverter for AnsiIntervalType") {
+    DataTypeTestUtils.yearMonthIntervalTypes.foreach { dt =>
+      val schema = new StructType().add(dt.typeName, dt)
+      val converter = new RowToColumnConverter(schema)
+      val columns = OnHeapColumnVector.allocateColumns(10, schema)
+      val batch = new ColumnarBatch(columns.toArray, 3)
```

Comment:
Where is `batch` used? and why it is needed?

Reply:
Sorry, a mistake. It should be `val batch = new ColumnarBatch(columns.toArray, **10**)`, It's meant to free up resources(`batch.close()`) instead of `columns.foreach(_.close())`

Reply:
Update.